### ISSUE TITLE
Convertion fails when url has query string

### ISSUE THREAD

**Issue by jcolano** [wkhtmltopdf/wkhtmltopdf#2750]
Hi,

We've tested this URL and it fails, as it is not recognizing the url parameters:

http://cloud.visitrack.com/?tkn=001EDDF0-05B5-40EA-A1BE-CDAB013D3CC8&guid=484de832-fd73-4495-9e03-7524899301cd&lan=en&format=pdf#customrpt/customReport

Any ideas?

Thanks!

Juan

**Comment by PhilterPaper** [comment]
Have you tried #2748?

**Comment by jcolano** [comment]
Hi,
Thanks for the reply!  Yes, I actually tried the "<url>" like so:
wkhtmltopdf "http://cloud.visitrack.com/?tkn=001EDDF0-05B5-40EA-A1BE-CDAB013D3CC8&guid=484de832-fd73-4495-9e03-7524899301cd&lan=en&format=pdf#customrpt/customReport" c:\docs\test.pdf

In the process I received this echo:
Loading pages (1/6)Counting pages (2/6)Resolving links (4/6)Loading headers and footers (5/6)Printing pages (6/6)Done

RESULT: The result was an empty pdf.

Then I test Google.com adding a query string equal to the one of my original URL, like so:
wkhtmltopdf "http://google.com/?tkn=001EDDF0-05B5-40EA-A1BE-CDAB013D3CC8&guid=484de832-fd73-4495-9e03-7524899301cd&lan=en&format=pdf#customrpt/customReport" c:\docs\test1.pdf
In the process I recelived this echo:
Loading pages (1/6)Counting pages (2/6)Resolving links (4/6)Loading headers and footers (5/6)Printing pages (6/6)Done
RESULT: The result was a PDF with Google's  page, so it worked (attached).
May be there's a timeout going on?
Thanks!!!
Juan

**Comment by bgadoury** [comment]
When I run this: 

`wkhtmltopdf --debug-javascript page "http://cloud.visitrack.com/?tkn=001EDDF0-05B5-40EA-A1BE-CDAB013D3CC8&guid=484de832-fd73-4495-9e03-7524899301cd&lan=en&format=pdf#customrpt/customReport" jcolano.pdf`

I get an empty PDF and the following console output:

`Warning: http://cloud.visitrack.com/js/app.js?bust=1452640677901:28 url --> http://cloud.visitrack.com`
`Warning: http://cloud.visitrack.com/js/app.js?bust=1452640677901:81 EXC customReport, fn: loadReport_Callback 2 :, exc: Unable to parse JSON string`

When I add an arbitrarily large --javascript-delay value like this:

`wkhtmltopdf --debug-javascript --javascript-delay 9000 page "http://cloud.visitrack.com/?tkn=001EDDF0-05B5-40EA-A1BE-CDAB013D3CC8&guid=484de832-fd73-4495-9e03-7524899301cd&lan=en&format=pdf#customrpt/customReport" jcolano.pdf`

I get a 4.4MB PDF and the following console output:

`Warning: http://cloud.visitrack.com/js/app.js?bust=1452640711157:28 url --> http://cloud.visitrack.com`

In summary, you need --javascript-delay to give that page enough time to do its thing. 9000ms is probably way more time than it actually needs, in fact.

**Comment by Tomsgu** [comment]
CLosing as solutions were provided.
